src/Microsoft.Azure.ServiceBus/Primitives/TaskExtensionHelper.cs

```diff
@@ -16,9 +17,9 @@ public static void Schedule(Func<Task> func)
                 {
                     await func().ConfigureAwait(false);
                 }
-                catch (Exception)
+                catch (Exception ex)
                 {
-                    // TODO: Log any unexpected exception here.
+                    MessagingEventSource.Log.ScheduleTaskFailed(func.Target.GetType().FullName, func.GetMethodInfo().Name, ex.ToString());
```

Comment:
Can you add a Non-Event and pass Exception as is and then the actual event which accepts the exception string